### PR TITLE
wayland-protocols: fix source references

### PR DESCRIPTION
**Pull request by maikelvdh**
Specify library name and version:  **wayland-protocols/***

It seems that the [wayland-project](https://github.com/wayland-project) organisation on GitHub is no longer having any public repositories. There all the source urls are no longer working. Unfortunately the GitLab setup is not having all the older versions listed in conandata available.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by jwillikers** [comment]
@maikelvdh Thanks for this PR. This is definitely an oversight that these weren't using the official GitLab project to get the source archives. Thanks for the update. You'll also need to remove the versions from the `config.yml` file.

**Comment by SpaceIm** [comment]
> Unfortunately the GitLab setup is not having all the older versions listed in conandata available.

I see these old versions (from 1.0 to 1.31): https://gitlab.freedesktop.org/wayland/wayland-protocols/-/tags

**Comment by maikelvdh** [comment]
> > > Unfortunately the GitLab setup is not having all the older versions listed in conandata available.
> > 
> > 
> > I see these old versions (from 1.0 to 1.31): https://gitlab.freedesktop.org/wayland/wayland-protocols/-/tags
> 
> That must be referring to the fact that there are only GitLab _Release_'s starting with version 1.27. Of course, there's still archive downloads available for versions before 1.27, i.e. `https://gitlab.freedesktop.org/wayland/wayland-protocols/-/archive/1.26/wayland-protocols-1.26.tar.bz2`.

Thanks for the references, I will add back the older releases then!

File: recipes/wayland-protocols/all/conandata.yml [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-    sha256: "04d3f66eca99d638ec8dbfdfdf79334290e22028f7d0b04c7034d9ef18a3248a"
+    url: "https://gitlab.freedesktop.org/wayland/wayland-protocols/-/archive/1.31/wayland-protocols-1.31.tar.bz2"
+    sha256: "3d7fa65a6f3c86747f780a628df04d3c7a981776dee2b6a492cc112b3bbe2820"
   "1.27":
-    url: "https://github.com/wayland-project/wayland-protocols/archive/1.27.tar.gz"
-    sha256: "6dd6ee86478adf4347f3b8b4f3da62dbe9e44912c9cef21cf99abfe692313df4"
-  "1.26":
-    url: "https://github.com/wayland-project/wayland-protocols/archive/1.26.tar.gz"
-    sha256: "fe56386f436a84e97c3b6a61b76306f205a64425900f247ad0048174b9c32d4d"
+    url: "https://gitlab.freedesktop.org/wayland/wayland-protocols/-/archive/1.27/wayland-protocols-1.27.tar.bz2"
```
Comment (SpaceIm) [inline recipes/wayland-protocols/all/conandata.yml:6]: Prefer official tarball for these 2 ones. It's always better than tarball generated on the fly by some service.

**Comment by roalter** [comment]
Saw this recently either. Should be placed upstream asap.

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**